### PR TITLE
Library version can now be extracted from header

### PR DESCRIPTION
**Pull request by stopiccot**

### CODE REVIEW

**Comment by stopiccot** [comment]
Pull request for #18

File: auto/src/glew_init_tail.c [hunk 1]
```diff
@@ -16,16 +16,15 @@ const GLubyte * GLEWAPIENTRY glewGetErrorString (GLenum error)
...
-  };
-  const size_t max_string = sizeof(_glewString)/sizeof(*_glewString) - 1;
-  return _glewString[(size_t)name > max_string ? 0 : (size_t)name];
+    case GLEW_VERSION:       return (const GLubyte*)"GLEW_VERSION_STRING";
+    case GLEW_VERSION_MAJOR: return (const GLubyte*)"GLEW_VERSION_MAJOR_STRING";
+    case GLEW_VERSION_MINOR: return (const GLubyte*)"GLEW_VERSION_MINOR_STRING";
+    case GLEW_VERSION_MICRO: return (const GLubyte*)"GLEW_VERSION_MICRO_STRING";
+  }
+
+  return NULL;
```
Comment (nigels-com) [inline auto/src/glew_init_tail.c]: I think some compilers are prone to complaining about any switch statement without a default clause.  If you don't mind shifting the NULL return into a default, I appreciate that.

File: auto/src/glew_tail.h [hunk 2]
```diff
@@ -8,10 +8,10 @@
...
 
 /* string codes */
-#define GLEW_VERSION 1
-#define GLEW_VERSION_MAJOR 2
-#define GLEW_VERSION_MINOR 3
-#define GLEW_VERSION_MICRO 4
+#define GLEW_VERSION GLEW_VERSION_MAJOR_STRINGGLEW_VERSION_MINOR_STRING
+#define GLEW_VERSION_MAJOR GLEW_VERSION_MAJOR_STRING
+#define GLEW_VERSION_MINOR GLEW_VERSION_MINOR_STRING
+#define GLEW_VERSION_MICRO GLEW_VERSION_MICRO_STRING
```
Comment (nigels-com) [inline auto/src/glew_tail.h]: I'm not in favour of these defines varying from one release to the next.  It's too fragile in terms of matching the header to a precompiled library.  I wonder if we could put the version information in a /* */ comment block in a manner that would be suitable for easy grepping, but would have no impact on the interface.

Comment (stopiccot) [inline auto/src/glew_tail.h]: Maybe you are right. I just haven't thought about such weird case when someone could use library version different from header version. It's really not a way to go. It's the same as pkg-config returning not matching version. Whole point is to check version from header at configure time so you can make some decisions in build scripts. And many open-source libraries seem to be ok defining version info like header constants. 

In the end of the day defining version info in comments can be an option. Honestly I don't like it but it's probably a matter of taste. Maybe we can introduce another 4 defines like a compromise. Anyway you are boss here it's up to you to decide. Just give us that ability to grep version from header.

Comment (nigels-com) [inline auto/src/glew_tail.h]: I'll take another look at this myself eventually, unless a fresh merge request appears. :-)

**Comment by nigels-com** [comment]
I put similar change into GLEW master branch just now.
It avoids touching any of the existing macros.

```$ grep '^VERSION ' include/GL/glew.h
VERSION 1.12.0

$ grep '^VERSION' include/GL/glew.h
VERSION 1.12.0
VERSION_MAJOR 1
VERSION_MINOR 12
VERSION_MICRO 0```

**Pull request closed without merging**